### PR TITLE
Adding a plugin to calculate the virtual temperature (version 2)

### PR DESCRIPTION
**Pull request by mo-philrelton**
Address: https://metoffice.atlassian.net/browse/EPPT-1965

This PR adds a plugin to calculate the virtual temperature based on the air temperature and humidity mixing ratio.

No CLI added due to the move to the DAG runner.

This is a replacement for the closed #2058 to avoid history rewrites and git configuration issues.

Testing:

Ran tests and they passed OK

    Added new tests for the new feature(s)

CLA

If a new developer, signed up to CLA

### CODE REVIEW

File: improver_tests/virtual_temperature/test_VirtualTemperature.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+"""Unit tests for the VirtualTemperature plugin"""
+
+from unittest.mock import patch, sentinel
+
+import numpy as np
+import pytest
+from iris.cube import Cube
+
+from improver.virtual_temperature import VirtualTemperature
+from improver_tests.utilities.copy_attributes.test_CopyAttributes import HaltExecution
```
Comment (Katie-Howard) [inline improver_tests/virtual_temperature/test_VirtualTemperature.py:14]: With the latest changes from a different ticket from  Marcus copy_attributes has been renamed copy_metadata so this now needs to be:
from improver_tests.utilities.copy_metadata.test_CopyMetadata import HaltExecution

**Review by Katie-Howard — CHANGES_REQUESTED** [review]
I think you just need to change a line in test_VirtualTemperature to get the tests to work and then it should be fine 👍

File: improver_tests/virtual_temperature/test_VirtualTemperature.py (hunk 1 above)
Comment (mo-philrelton) [reply improver_tests/virtual_temperature/test_VirtualTemperature.py]: Done

**Review by Katie-Howard — DISMISSED** [review]
I'm happy with these changes now 👍

**Review by mo-robertneal — DISMISSED** [review]
The code changes look good to me and the unit tests all passed.

**Review by Katie-Howard — APPROVED** [review]
The failing test has been resolved so I'm happy with this 👍

**Pull request merged**